### PR TITLE
Context Action Menu

### PR DESCRIPTION
**Pull request by falko17**
This implements the context action menu, which can be triggered by right-clicking a node or an edge.
Right-clicking an element within the TreeView window also works.

## Available actions
The following actions have been implemented:
* Delete node/edge
* Show properties
* Show code
* Show in tree view (node only)
* Show at source node / target node in tree view (edge only)
* Show unified diff (clone edge only)
* Accept divergence (divergent reflexion edge only)

## Additional changes
In addition to the context action menu, the following has been implemented:
* `PopupMenu` and `PopupMenuAction` are general UI components that may be used for other menus than the context action menu.
* Issue #657 has been fixed.
* Issue #660 has been fixed.
* The TreeView window will now automatically rebuild whenever the corresponding graph is modified.
* A pre-commit hook has been added that updates the Unity version number across the entire project whenever it is changed in the `ProjectSettings/ProjectVersion.txt`.
* A bad pattern has been added for `TODO`s and `FIXME`s without an issue number associated to it. For example, `// TODO: Something` triggers this check, as it should be something like `// TODO (#632): Something` instead. See the comments from github-actions below for an example. 
* Updated to Unity 2022.3.13f1.
* Added a screenshot of SEE to the README file.
* The name of the artificial root node has been changed from a random ID to `GraphName#ROOT`.
* The `check_for_bad_patterns` script has been updated to use alerts instead of emojis.
* Some minor fixes and code style improvements in various places.

### CODE REVIEW

**Comment by falko17** [comment]
Tests are failing because the docker image for Unity 2022.3.13f1 does not exist yet. This should get resolved in a few days, the branch is ready to be reviewed though.

**Review by koschke — COMMENTED** [review]
I had only two very, very minor comments. There were so minor that I fixed them myself.

File: Assets/SEE/UI/Window/TreeWindow/TreeWindow.cs [hunk 1]
```diff
@@ -54,12 +59,54 @@ public partial class TreeWindow : BaseWindow
...
+                AddNode(root);
+            }
+
+            if (roots.Count == 0)
+            {
+                ShowNotification.Warn("Empty graph", "Graph has no roots. TreeView will be empty.");
+            }
+        }
+
+
```
[the remark this answers is not in the document]
Comment (koschke) [reply Assets/SEE/UI/Window/TreeWindow/TreeWindow.cs]: I fixed it.

File: Assets/SEE/UI/PopupMenu/PopupMenu.cs [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        /// These actions will be added to the menu once it is started.
+        /// </summary>
+        private readonly Queue<PopupMenuAction> ActionsBeforeStart = new();
+
+        /// <summary>
+        /// Whether the menu should currently be shown.
+        /// </summary>
+        private bool ShouldShowMenu;
+
+        private float menuHeight => Menu.sizeDelta.y;
```
[the remark this answers is not in the document]
Comment (koschke) [reply Assets/SEE/UI/PopupMenu/PopupMenu.cs:51]: I fixed it.

**Review by koschke — APPROVED** [review]
Names of properties shall start with a capital letter. I fixed it.

**Pull request merged**